### PR TITLE
DROID-1618 Set | Enhancement | Layout menu

### PR DESCRIPTION
**Pull request by konstantiniiv**

### CODE REVIEW

File: core-ui/src/main/res/drawable/ic_arrow_disclosure_18.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="18dp"
+    android:height="18dp"
+    android:viewportWidth="18"
+    android:viewportHeight="18">
+  <path
+      android:pathData="M6.22,14.789C5.927,14.508 5.927,14.052 6.22,13.771L11.189,9L6.22,4.229C5.927,3.948 5.927,3.492 6.22,3.211C6.513,2.93 6.987,2.93 7.28,3.211L13.311,9L7.28,14.789C6.987,15.07 6.513,15.07 6.22,14.789Z"
+      android:fillColor="#A7A7A7"
```
Comment (uburoiubu) [inline core-ui/src/main/res/drawable/ic_arrow_disclosure_18.xml:8]: palette color?

File: core-ui/src/main/res/drawable/ic_layout_gallery.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <item android:offset="1" android:color="#00F2F2F2"/>
+      </gradient>
+    </aapt:attr>
+  </path>
+  <path
+      android:pathData="M13,13H23C24.105,13 25,13.895 25,15V23C25,24.105 24.105,25 23,25H13C11.895,25 11,24.105 11,23V15C11,13.895 11.895,13 13,13ZM10,15C10,13.343 11.343,12 13,12H23C24.657,12 26,13.343 26,15V23C26,24.657 24.657,26 23,26H13C11.343,26 10,24.657 10,23V15ZM13,29H23C24.105,29 25,29.895 25,31V39C25,40.105 24.105,41 23,41H13C11.895,41 11,40.105 11,39V31C11,29.895 11.895,29 13,29ZM10,31C10,29.343 11.343,28 13,28H23C24.657,28 26,29.343 26,31V39C26,40.657 24.657,42 23,42H13C11.343,42 10,40.657 10,39V31ZM41,13H31C29.895,13 29,13.895 29,15V23C29,24.105 29.895,25 31,25H41C42.105,25 43,24.105 43,23V15C43,13.895 42.105,13 41,13ZM31,12C29.343,12 28,13.343 28,15V23C28,24.657 29.343,26 31,26H41C42.657,26 44,24.657 44,23V15C44,13.343 42.657,12 41,12H31ZM31,29H41C42.105,29 43,29.895 43,31V39C43,40.105 42.105,41 41,41H31C29.895,41 29,40.105 29,39V31C29,29.895 29.895,29 31,29ZM28,31C28,29.343 29.343,28 31,28H41C42.657,28 44,29.343 44,31V39C44,40.657 42.657,42 41,42H31C29.343,42 28,40.657 28,39V31Z"
+      android:fillColor="#A7A7A7"
```
Comment (uburoiubu) [inline core-ui/src/main/res/drawable/ic_layout_gallery.xml:24]: idem

File: core-ui/src/main/res/drawable/ic_layout_grid.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          android:endY="30.928"
+          android:type="linear">
+        <item android:offset="0" android:color="#FFE9E9E9"/>
+        <item android:offset="1" android:color="#00F2F2F2"/>
+      </gradient>
+    </aapt:attr>
+  </path>
+  <path
+      android:pathData="M22.999,13H41C42.105,13 43,13.895 43,15V39C43,40.105 42.105,41 41,41H22.999V33H42.999V32H22.999V22.5H42.999V21.5H22.999V13ZM21.999,13H13C11.895,13 11,13.895 11,15V21.5H21.999V13ZM11,22.5V32H21.999V22.5H11ZM11,33V39C11,40.105 11.895,41 13,41H21.999V33H11ZM10,15C10,13.343 11.343,12 13,12H41C42.657,12 44,13.343 44,15V39C44,40.657 42.657,42 41,42H13C11.343,42 10,40.657 10,39V15Z"
+      android:fillColor="#A7A7A7"
```
Comment (uburoiubu) [inline core-ui/src/main/res/drawable/ic_layout_grid.xml:23]: idem

File: core-ui/src/main/res/drawable/ic_list_arrow_18.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="18dp"
+    android:height="18dp"
+    android:viewportWidth="18"
+    android:viewportHeight="18">
+  <path
+      android:pathData="M4,7L9,12L14,7"
+      android:strokeWidth="1.5"
+      android:fillColor="#00000000"
+      android:strokeColor="#A7A7A7"
```
Comment (uburoiubu) [inline core-ui/src/main/res/drawable/ic_list_arrow_18.xml:10]: idem

File: core-ui/src/main/res/drawable/ic_layout_gallery.xml (hunk 2 above)
Comment (konstantiniiv) [reply core-ui/src/main/res/drawable/ic_layout_gallery.xml:24]: fixed

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerLayoutWidgetUi.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+import com.anytypeio.anytype.core_models.Block
+import com.anytypeio.anytype.core_models.DVViewer
+import com.anytypeio.anytype.core_models.DVViewerType
+import com.anytypeio.anytype.core_models.Id
+import com.anytypeio.anytype.core_models.Relations
+import com.anytypeio.anytype.domain.objects.StoreOfRelations
+
+data class ViewerLayoutWidgetUi(
+    val viewer: Id?,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerLayoutWidgetUi.kt:11]: is it a valid state when viewer id is null? if it is not, then `Id` should not be nullable here.

File: core-ui/src/main/res/drawable/ic_layout_grid.xml (hunk 3 above)
Comment (konstantiniiv) [reply core-ui/src/main/res/drawable/ic_layout_grid.xml:23]: done

File: core-ui/src/main/res/drawable/ic_list_arrow_18.xml (hunk 4 above)
Comment (konstantiniiv) [reply core-ui/src/main/res/drawable/ic_list_arrow_18.xml:10]: done

File: core-ui/src/main/res/drawable/ic_arrow_disclosure_18.xml (hunk 1 above)
Comment (konstantiniiv) [reply core-ui/src/main/res/drawable/ic_arrow_disclosure_18.xml]: fixed

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerLayoutWidgetUi.kt (hunk 5 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerLayoutWidgetUi.kt:11]: In the case when the DataView's View has not arrived yet, either null or "" should be set here.

**Review by uburoiubu — APPROVED** [review]

**Pull request merged**